src/CMakeLists.txt

```diff
@@ -186,6 +186,10 @@ if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
   set(TOIT_BLUETOOTH_LIBS "-framework Foundation" "-framework CoreBluetooth" ObjC)
 endif()
 
+if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin" OR (${CMAKE_SYSTEM_NAME} MATCHES "Linux" AND NOT CMAKE_SIZEOF_VOID_P EQUAL 4))
```

Comment:
This should probably use `FindZLIB` with `ZLIB_USE_STATIC_LIBS` to `ON`.
https://cmake.org/cmake/help/latest/module/FindZLIB.html

The way it's written now, it will use the zlib library if it finds it, and otherwise ignore it.
You can force the inclusion with `find_package(ZLIB REQUIRED)`.
On Linux64 the `ZLIB_US_STATIC_LIBS` is probably not needed either.

On the buildbot we should now ensure that the Windows bot has a static version of the zlib.

``` diff
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c8195a3a..d8f3a6cd 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -13,6 +13,8 @@
 # The license can be found in the file `LICENSE` in the top level
 # directory of this repository.
 
+cmake_minimum_required(VERSION 3.23.0)
+
 if (NOT "${TOIT_SYSTEM_NAME}" MATCHES "esp32")
   add_custom_target(build_tools)
   add_subdirectory(compiler)
@@ -186,6 +188,13 @@ if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
   set(TOIT_BLUETOOTH_LIBS "-framework Foundation" "-framework CoreBluetooth" ObjC)
 endif()
 
+set(ZLIB_USE_STATIC_LIBS "ON")
+find_package(ZLIB)
+if (ZLIB_FOUND)
+  include_directories(${ZLIB_INCLUDE_DIRS})
+  set(TOIT_LINK_LIBS ${TOIT_LINK_LIBS} ${ZLIB_LIBRARIES})
+endif(ZLIB_FOUND)
+
 # Because of the `CACHE INTERNAL ""` at the end of the `set` we can
 #   use this variable outside of the directory.
 set(TOIT_LINK_LIBS
```